docs-starlight/src/content/docs/04-reference/01-hcl/02-blocks.mdx

```diff
@@ -1715,3 +1716,100 @@ inputs = {
+    vpc_name = "main"
+    cidr     = "10.0.0.0/16"
+  }
+}
+
+unit "database" {
+  source = "git::git@github.com:acme/infrastructure-units.git//database/rds?ref=v0.0.1"
+  path   = "rds"
+  values = {
+    engine   = "postgres"
+    version  = "13"
+  }
+  no_dot_terragrunt_stack = true
+}
+```
+
+With the above configuration, the resulting directory structure will be:
+
+<FileTree>
+
+- terragrunt.stack.hcl
+- .terragrunt-stack
+  - vpc
+    - terragrunt.values.hcl
+    - terragrunt.hcl
+- rds
+  - terragrunt.values.hcl
+  - terragrunt.hcl
+
+</FileTree>
+
+The `vpc` unit is placed inside `.terragrunt-stack`, as expected.
+The `database` (`rds`) unit is generated in the **same directory as `terragrunt.stack.hcl`**, rather than inside `.terragrunt-stack`, due to `no_dot_terragrunt_stack = true`.
+
+**Note:**
+The `source` value can be updated dynamically using the `--source-map` flag, just like `terraform.source`.
+
+## stack
+
+<Aside type="tip" title="Stacks">
+    The <code dir="auto">stack</code> block is experimental, usage requires the <a href="/docs/reference/experiments#stacks"><code dir="auto">--experiment stacks</code></a> flag.
+</Aside>
+
+
+The `stack` block is used to define a stack of deployment units in a Terragrunt configuration file (`terragrunt.stack.hcl`).
+Stacks allow for nesting, enabling the organization of infrastructure components into modular, reusable groups, reducing redundancy and improving maintainability.
+
+Stacks are designed to be nestable, helping to mitigate the risk of stacks becoming too large or too repetitive.
+When a stack is generated, it can include nested stacks, ensuring that the configuration scales efficiently.
+
+The `stack` block supports the following arguments:
+
+- `name` (label): A unique identifier for the stack. This is used to reference the stack elsewhere in your configuration.
+- `source` (attribute): Specifies where to find the Terragrunt configuration files for this stack. This follows the same syntax as the `source` parameter in the `terraform` block.
+- `path` (attribute): The relative path within `.terragrunt-stack` where this stack should be generated.If an absolute path is provided here, Terragrunt will generate the stack in that location, instead of generating it in a path relative to the `.terragrunt-stack` directory.
+- `values` (attribute, optional): A map of custom values that can be passed to the stack. These values can be referenced within the stack's configuration files, allowing for customization without modifying the stack source.
+- `no_dot_terragrunt_stack` (attribute, optional): A boolean flag (`true` or `false`). When set to `true`, the stack **will not** be placed inside `.terragrunt-stack` but will instead be generated in the **same directory as `terragrunt.stack.hcl`**.
+This enables **soft adoption** of stacks, allowing users to introduce `terragrunt.stack.hcl` above existing directories without restructuring their project layout.
+
+Example:
+
+```hcl
```

Comment:
I'd recommend splitting this into two code-fences to make it easier to see what's happening here.